core/src/main/kotlin/com/bradley/wilson/core/functional/Either.kt

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright (C) 2019 Fernando Cejas Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.bradley.wilson.core.functional
+
+import com.bradley.wilson.core.functional.Either.Left
+import com.bradley.wilson.core.functional.Either.Right
+
+/**
+ * Represents a value of one of two possible types (a disjoint union).
+ * Instances of [Either] are either an instance of [Left] or [Right].
+ * FP Convention dictates that [Left] is used for "failure"
+ * and [Right] is used for "success".
+ *
+ * @see Left
+ * @see Right
+ */
+sealed class Either<out L, out R> {
```

Comment:
Bringing a very popular monad from the functional programming sphere is key to the error handling aspect of this project. It's key because we are using coroutines, which doesn't become too strict on how we determine errors from downstream to upstream, it's not tightly coupled with any given approach moreover it's determines how you make these requests and that's all unlike other async alternatives such as rxJava/Kotlin.